src/Sulu/Bundle/AdminBundle/Controller/WidgetGroupsController.php

```diff
@@ -36,14 +36,14 @@ public function groupAction($groupAlias, Request $request)
         try {
             $groupAlias = str_replace('-', '_', $groupAlias);
 
-            return new Response(
-                $this->getWidgetsHandler()->renderWidgetGroup(
-                    $groupAlias,
-                    $request->query->all()
-                )
+            $content = $this->getWidgetsHandler()->renderWidgetGroup(
+                $groupAlias,
+                $request->query->all()
             );
+
+            return new Response($content, $content !== '' ? 200 : 204);
         } catch (WidgetException $ex) {
-            return new Response($ex->getMessage());
+            return new Response($ex->getMessage(), 404);
```

Comment:
Is 404 really the correct error code here? Is the only error that can appear that the widget doesn't exist? Then I would rename the exception to `WidgetNotFoundException` or even create a new sub exception.